### PR TITLE
Hw 2

### PR DESCRIPTION
**Pull request by gigbat**

### CODE REVIEW

File: src/main/java/core/basesyntax/RightTriangle.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    public RightTriangle(String color, int side) {
+        super(color);
+        this.side = side;
+        draw();
+    }
+
+    @Override
+    public void draw() {
+        area = this.side * this.side / 2;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/RightTriangle.java:14]: why do we calculate area in the draw method?

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Circle.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    int radius;
```
[the remark this answers is not in the document]
Comment (gigbat) [reply src/main/java/core/basesyntax/Circle.java:4]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public abstract class Figure {
+    String color;
```
[the remark this answers is not in the document]
Comment (gigbat) [reply src/main/java/core/basesyntax/Figure.java:4]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    double area;
+
+    public Figure(String color) {
+        this.color = color;
+    }
+
+    public abstract void draw();
```
[the remark this answers is not in the document]
Comment (gigbat) [reply src/main/java/core/basesyntax/Figure.java:11]: done

File: src/main/java/core/basesyntax/Demo.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Demo {
+    public static void main(String[] args) {
```
[the remark this answers is not in the document]
Comment (gigbat) [reply src/main/java/core/basesyntax/Demo.java:4]: done

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
```
[the remark this answers is not in the document]
Comment (gigbat) [reply src/main/java/core/basesyntax/IsoscelesTrapezoid.java:3]: done

File: src/main/java/core/basesyntax/RightTriangle.java (hunk 1 above)
Comment (gigbat) [reply src/main/java/core/basesyntax/RightTriangle.java]: done

**Comment by RomanDubovskyi** [comment]
let's try to make commit messages a bit more informative 🙂
image

File: src/main/java/core/basesyntax/Demo.java [hunk 7]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Demo {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[new Random().nextInt(5)];
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Demo.java:7]: declare 5 as a constant, now it's a magic number (not clear why 5 and not 6, etc)

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 8]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+    public Figure get() {
+        int index = new Random().nextInt(5);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:7]: same here with magic number

File: src/main/java/core/basesyntax/Rectangle.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: rectangle, area: " + getArea() + " sq. units, diagonal: "
+                + diagonal + ", height: " + height
+                + ", weight: " + weight + ", color: " + getColor());
+    }
+
+    @Override
+    public void calculate() {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java:23]: method comletes 2 tasks at the same time, let's splite the login into to separate methods

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/Demo.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Demo {
+    static final int sizeOfArrayForFigures = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Demo.java:6]:
```suggestion
     private static final int MAX_ARR_SIZE = 5;
```
don't forget about access modifiers + we use upper case to declare constant

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+    static final int MAGIC_NUMBER = 5;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:6]: same with access modifier

**Review by RomanDubovskyi — APPROVED** [review]
add access modifiers, rest is good

**Pull request closed without merging**